packages/contracts/src/UnirepContract.ts

```diff
@@ -0,0 +1,134 @@
+import path from 'path'
+import { ethers } from 'ethers'
+import { CircuitName } from '@unirep/circuits'
+
+import { UnirepABI } from './abis/Unirep'
+import { Unirep } from '../src/contracts'
+import { ContractConfig } from './types/config'
+
+export default class UnirepContract {
+    private static createVerifierName = (circuit: string) => {
+        return `${circuit.charAt(0).toUpperCase() + circuit.slice(1)}Verifier`
+    }
+
+    private static getUnirepArtifact = (
+        artifactsPath: string
+    ): { abi; bytecode } => {
+        const Unirep = require(path.join(
+            artifactsPath,
+            'contracts',
+            'Unirep.sol',
+            'Unirep.json'
+        ))
+        return Unirep
+    }
+
+    /**
+     * Deploy Unirep contract and verifiers with compiled artifacts
+     * @param artifactsPath The path to the artifacts (e.g. ../build/artifacts/)
+     * @param deployer The deployer of the contract
+     * @param config The config of the contracts and its circuits
+     * @returns The Unirep contract object
+     */
+    public static deploy = async (
```

Comment:
I think most app developers don't need this function. Really it should be a script intended to be run on a dev machine.